tests/lib/nested.sh

```diff
@@ -1462,19 +1462,24 @@ nested_start_core_vm_unit() {
 }
 
 nested_setup_vm(){
-
     if [ "${SNAPD_USE_PROXY:-}" = true ]; then
         nested_no_proxy="${NO_PROXY},10.0.2.2"
 
         # Ensure the nameservers used are the same than the host vm
-        net_interface="$(ip route show default | awk '{print $5}')"
-        nameservers="$(resolvectl status "$net_interface" | grep "DNS Servers:" | cut -d: -f2)"
-        if [ -n "$nameservers" ]; then
-            remote.exec "grep -v '^nameserver' /etc/resolv.conf > /tmp/resolv.conf"
-            remote.exec "sudo cp /tmp/resolv.conf /etc/resolv.conf"
-            for nameserver in $nameservers; do
-                remote.exec "echo nameserver $nameserver | sudo tee -a /etc/resolv.conf"
-            done
+        if os.query is-ubuntu-ge 18.04; then
+            net_interface="$(ip route show default | awk '{print $5}')"
+            nameservers="$(resolvectl status "$net_interface" | grep "DNS Servers:" | cut -d: -f2)"
+            if [ -n "$nameservers" ]; then
+                remote.exec "grep -v '^nameserver' /etc/resolv.conf > /tmp/resolv.conf"
+                remote.exec "sudo cp /tmp/resolv.conf /etc/resolv.conf"
+                for nameserver in $nameservers; do
+                    remote.exec "echo nameserver $nameserver | sudo tee -a /etc/resolv.conf"
+                done
```

Comment:
Nitpick: might as well avoid sudo while building the new resolv.conf?

```suggestion
                for nameserver in $nameservers; do
                    remote.exec "echo nameserver $nameserver >> /etc/resolv.conf"
                done
                remote.exec "sudo cp /tmp/resolv.conf /etc/resolv.conf"
```